scripts/tasks_retrival/gene_disease_association_task_creation.py

```diff
@@ -0,0 +1,275 @@
+        which holds the gene ids and return the mapping dataframe.
+        this is a set of unique mappings.  It is used to map the ids in the data,
+        which will contain duplicates, to the corresponding symbol names
+    Args:
+        list_of_gene_metadata (list): list containing gene metadata.
+
+    Returns
+    -------
+        pd.DataFrame: a data frame with the gene id as index with the symbol as value
+
+    """
+    gene_metadata_df = pd.DataFrame(list_of_gene_metadata)
+    # some target ids have multiple symbols
+    gene_metadata_df = gene_metadata_df.drop_duplicates(subset="query")
+    gene_metadata_df.index = gene_metadata_df["query"]
+    return gene_metadata_df
+
+
+def get_symbols(gene_targetId_list):
+    """
+        given s list of gene id's (names Like ENSG00000006468) this method
+        uses the MyGenInfo package to retrieve the gene symbol (name like PLAC4).
+        The retreaved dataframe is made into a translation df from id to metadata (symbol in this case),
+        and is used to translate the original list ot a symbols of the same length, with the matching symbols
+        in the corresponding locations.
+
+    Args:
+    ----
+        gene_targetId_list (list): list of gene id's (names Like ENSG00000006468)
+
+    Returns:
+    -------
+         pd.DataFrame: dataframe with symbols corresponding to the input
+
+    """
+    mg = mygene.MyGeneInfo()
+    list_of_gene_metadata = mg.querymany(
+        gene_targetId_list.drop_duplicates(), species="human", fields="symbol"
+    )
+    gene_metadata_df = get_id_to_symbol_df(list_of_gene_metadata)
+    return pd.DataFrame(
+        [gene_metadata_df.loc[x, "symbol"] for x in gene_targetId_list],
+        columns=["symbol"],
+    )
+
+
+def get_gene_drug_assosiation_data(input_file: str):
+    """
+    read data from open-targets' parquet data files, either directly or from a local file copy.
+    The files are arranged with a fixed base name and part-index going from 0 to 199.  This is a standard way to save such data in parts
+    and the total number of parts differes, so the code starts at 0 until it fails to read the file, and stops there.
+
+    Args:
+    ----
+        input_file (str): base path (url or file path) of the parquet directory
+
+    Raises:
+    ------
+        RuntimeError: if no files had could be loaded
+
+    Returns:
+    -------
+        pd.Dataframe: concatinated dataframe containing all the loaded data from all files
+
+    """
+    res = []
+    file_exist = True
+    part_ind = 0
+    while file_exist:
+        try:
+            link_ars = f"associationByDatasourceDirect/part-{part_ind:05}-6866be1a-be5d-40cf-bdf6-627bef1d0410-c000.snappy.parquet"
+            gda_df = pd.read_parquet(input_file + link_ars)
+            res.append(gda_df)
+            part_ind = part_ind + 1
+        except Exception as exception:
+            if not len(res):
+                raise RuntimeError(f"could not read from {input_file}") from exception
```

Comment:
If the file count starts from 1, this will raise an error, but files might still exist. Locating the if clause after the loop will make the error correspond to the documentation.

Reply:
I am sorry, I don't understand.  once the except is fired, the loop is over.  Can you explain how this may give the wrong result?

Reply:
So, this error is to be fired if there is no data. But if the data files start from part int:=1, what will happen is that the method will search for part_ind==0, will not find it, see that  `not len(res)` is true, and an exception will be fired, but there is data...
So, that's exactly the problem: We will get out of the loop when we don't necessarily need to.